gdal/port/cpl_vsil.cpp

```diff
@@ -437,7 +437,7 @@ void VSICloseDir(VSIDIR* dir)
  * \brief Create a directory.
  *
  * Create a new directory with the indicated mode.  The mode is ignored
- * on some platforms.  A reasonable default mode value would be 0666.
+ * on some platforms.  A reasonable default mode value would be 0777.
```

Comment:
You generally don't want to give write permissions to group and other, so:

```suggestion
 * on some platforms.  A reasonable default mode value would be 0755.
```

Reply:
No, I often don't want to give write permissions to group and other. And sometimes I even don't want to give any permissions to other. But this is normally well resolved by the mode given to `mkdir` being "modified by the process' file creation mask" aka `umask`: 
https://linux.die.net/man/3/mkdir 
http://manpages.ubuntu.com/manpages/focal/en/man2/mkdir.2.html 

So explicitly: IMO 0777 is right for Linux and POSIX. But if 0777 is unsafe with regard to GDAL's different target OS or VSI implementation, 0755 maybe a better recommendation in GDAL's documentation.

Reply:
> No, I often don't want to give write permissions to group and other

I'm a bit confused by your message. 0755 is exactly that: rwx for user, rx for group and other, so it is more restrictive than 0777 and "safer"

Reply:
Even more explicitly: 

`mkdir("files", 0777)` does *not* mean: Create dir "files" with exact permissions 0777. 
What it really means is: Create dir "files" with the result of `0777 & ~umask`.

`mkdir("files", 0755)` does *not* mean: Create dir "files" with exact permissions 0755. 
What it really means is: Create dir "files" with the result of `0755 & ~umask`.

A typical umask is 0022, i.e. don't give write permissions to group and others. Only if I explicitly change umask to 0000, files and dirs can be created with write permissions for group and other. And this is exactly what I expect when I change umask to 0000.

So in order for a program to behave correctly in the context of `umask`, you have to use `mkdir("file", 0777)`.

Reply:
(Disclaimer: I'm not an expert on low-level functions.)